vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -980,46 +981,72 @@ public HierarchicalDataProvider<T, SerializablePredicate<T>> getDataProvider() {
+     * {@link HierarchyFormat#FLATTENED}: the index refers to an item in the
+     * entire flattened tree, not only the root level, allowing items at any
+     * expanded level to be reached with this method.
+     * <p>
+     * If the index exceeds the available range, scrolling goes to the last item
+     * according to the current hierarchy format.
      *
-     * @param rowIndex
-     *            zero based index of the item in the root level of the tree
-     * @see TreeGrid#scrollToIndex(int...)
+     * @param index
+     *            zero based index of the item to scroll to
      */
     @Override
-    public void scrollToIndex(int rowIndex) {
+    public void scrollToIndex(int index) {
         getUI().ifPresent(
                 ui -> ui.beforeClientResponse(this, ctx -> getElement()
-                        .executeJs("this.scrollToIndex($0);", rowIndex)));
+                        .executeJs("this.scrollToIndex($0);", index)));
     }
 
     /**
-     * Scrolls to a nested item within the tree.
+     * Scrolls to a nested item specified by its hierarchical path.
+     * <p>
+     * The hierarchical path is an array of indexes where each index refers to a
+     * child of the item at the previous index. Scrolling continues until it
+     * reaches the last index in the array or encounters a collapsed item.
      * <p>
-     * The `indexes` parameter can be either a single number or multiple
-     * numbers. The grid will first try to scroll to the item at the first index
-     * in the root level of the tree. In case the item at the first index is
-     * expanded, the grid will then try scroll to the item at the second index
-     * within the children of the expanded first item, and so on. Each given
-     * index points to a child of the item at the previous index.
+     * For example, given {@code &#123; 2, 1, ... &#125;} as the path, the
+     * component will first try to scroll to the item at index 2 in the root
+     * level. If that item is expanded, it will then try to scroll to the item
+     * at index 1 among its children, and so forth.
+     * <p>
+     * This method is supported only for data providers that use
+     * {@link HierarchyFormat#NESTED}. For {@link HierarchyFormat#FLATTENED},
+     * use {@link #scrollToIndex(int)} with a flat index instead.
      *
-     * @param indexes
-     *            zero based row indexes to scroll to
+     * @param path
+     *            zero based row path to scroll to
+     * @throws UnsupportedOperationException
+     *             if the data provider uses a hierarchy format other than
+     *             {@link HierarchyFormat#NESTED}
      * @see TreeGrid#scrollToIndex(int)
      */
-    public void scrollToIndex(int... indexes) {
-        if (indexes.length == 0) {
+    public void scrollToIndex(int... path) {
+        if (!getDataProvider().getHierarchyFormat()
+                .equals(HierarchyFormat.NESTED)) {
+            throw new UnsupportedOperationException(
```

Comment:
Is there a test for this exception already? If not, let's either add it or keep this PR only about JavaDoc.

Reply:
I've removed this change and will add it through a separate PR.